great_expectations/datasource/datasource_dict.py

```diff
@@ -70,10 +70,7 @@ def data(self) -> dict[str, FluentDatasource]:  # type: ignore[override] # `data
 
         configs = self._datasource_store.get_all()
         for config in configs:
-            if isinstance(config, FluentDatasource):
-                name = config.name
```

Comment:
These checks are just confusing at this point - removing as we should only have FDS